presto-main/src/main/java/com/facebook/presto/metadata/InternalSignatureUtils.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.metadata;
+
+import com.facebook.presto.spi.function.OperatorType;
+import com.facebook.presto.spi.function.Signature;
+import com.facebook.presto.spi.function.SignatureUtils;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeSignature;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static com.facebook.presto.spi.function.FunctionKind.SCALAR;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.unmodifiableList;
+import static java.util.stream.Collectors.toList;
+
+public final class InternalSignatureUtils
```

Comment:
Why do we need to move this one into SPI? Is it just because I haven't finished the refactoring? I don't think any of the `internalOperator` functions need to be available to plugins. After I finish the refactoring with scalar functions nor did any of the `internalScalarFunction`.

Reply:
This file will stay in presto-main. We are not moving it.

Reply:
Sorry read it wrong.